OpenRA.Mods.Common/Traits/Attack/AttackBase.cs

```diff
@@ -279,13 +279,21 @@ public AttackOrderTargeter(AttackBase ab, int priority, bool negativeDamage)
 			public int OrderPriority { get; private set; }
 			public bool OverrideSelection { get { return true; } }
 
-			bool CanTargetActor(Actor self, Target target, TargetModifiers modifiers, ref string cursor)
+			bool CanTargetActor(Actor self, Target target, ref TargetModifiers modifiers, ref string cursor)
 			{
 				IsQueued = modifiers.HasModifier(TargetModifiers.ForceQueue);
 
 				if (modifiers.HasModifier(TargetModifiers.ForceMove))
 					return false;
 
+				// Disguised actors are revealed by the attack cursor
+				// HACK: works around limitations in the targeting code that force the
+				// targeting and attacking logic (which should be logically separate)
+				// to use the same code
+				if (target.Actor.EffectiveOwner != null && target.Actor.EffectiveOwner.Disguised &&
```

Comment:
Isn't this (`target.Actor`) a null-reference risk if the target is a frozen actor?